File: frontend/lib/models/venue.dart
class Venue {
  const Venue({
    required this.id,
    required this.name,
    required this.area,
    required this.priceBand,
    required this.lat,
    required this.lng,
    required this.description,
    required this.tagIds,
    this.imageUrl,
    this.imageUrls = const [],
    this.mapsUrl,
  });

  final String id;
  final String name;
  final String area;
  final String priceBand;
  final double lat;
  final double lng;
  final String description;
  final List<String> tagIds;
  final String? imageUrl;
  final List<String> imageUrls;
  final String? mapsUrl;

  List<String> get galleryImages {
    if (imageUrls.isNotEmpty) return imageUrls;
    if (imageUrl != null && imageUrl!.isNotEmpty) return [imageUrl!];
    return const [];
  }

  factory Venue.fromJson(Map<String, dynamic> json) {
    final urls = (json['image_urls'] as List<dynamic>? ?? const [])
        .map((url) => url as String)
        .where((url) => url.isNotEmpty)
        .toList();
    final primary = json['image_url'] as String?;

    return Venue(
      id: json['id'] as String,
      name: json['name'] as String,
      area: json['area'] as String,
      priceBand: json['price_band'] as String,
      lat: (json['lat'] as num).toDouble(),
      lng: (json['lng'] as num).toDouble(),
      description: json['description'] as String? ?? '',
      tagIds: (json['tag_ids'] as List<dynamic>? ?? const [])
          .map((id) => id as String)
          .toList(),
      imageUrl: primary,
      imageUrls: urls.isNotEmpty
          ? urls
          : primary != null && primary.isNotEmpty
              ? [primary]
              : const [],
      mapsUrl: json['maps_url'] as String?,
    );
  }
}

File: backend/app/schemas/venue.py
from pydantic import BaseModel, Field

from app.schemas.common import PriceBand


class VenueBase(BaseModel):
    name: str = Field(min_length=2, max_length=120)
    area: str = Field(min_length=2, max_length=80)
    lat: float = Field(ge=-90, le=90)
    lng: float = Field(ge=-180, le=180)
    description: str = Field(default="", max_length=1200)
    tag_ids: list[str] = Field(default_factory=list)
    price_band: PriceBand
    image_url: str | None = Field(default=None, max_length=500)
    image_urls: list[str] = Field(default_factory=list, max_length=12)
    maps_url: str | None = Field(default=None, max_length=500)


class VenueCreate(VenueBase):
    pass


class VenueResponse(VenueBase):
    id: str


class VenueUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=120)
    area: str | None = Field(default=None, min_length=2, max_length=80)
    lat: float | None = Field(default=None, ge=-90, le=90)
    lng: float | None = Field(default=None, ge=-180, le=180)
    description: str | None = Field(default=None, max_length=1200)
    tag_ids: list[str] | None = None
    price_band: PriceBand | None = None
    image_url: str | None = Field(default=None, max_length=500)
    image_urls: list[str] | None = None
    maps_url: str | None = Field(default=None, max_length=500)


class VenueFilter(BaseModel):
    location: str | None = None
    product_tags: list[str] = Field(default_factory=list)
    vibe_tags: list[str] = Field(default_factory=list)
    price_band: PriceBand | None = None
